core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -52,8 +55,41 @@ private[spark] sealed trait MapStatus {
    * partitionId of the task or taskContext.taskAttemptId is used.
    */
   def mapId: Long
+
 }
 
+private[spark] class MapStatusLocationFactory(conf: SparkConf) {
```

Comment:
Shall we put this into `MapStatusLocationFactory.scala`?

Reply:
Sure, I was thinking of that too.